### ISSUE TITLE
Last release, compute errors with terascale cards.

### ISSUE THREAD

**Issue by guile124** [fireice-uk/xmr-stak#1474]
It's working the last xmr stak release (2.4.3) for AMD 69xxx old cards (Terascale)? I try to mine new lite v7 and Crypt-heavy with the same config and only get COMPUTE ERROR's

**Comment by Spudz76** [comment]
Running on HD5570 (Redwood) shows 37 bad results in 37 hours at 77H/s

**Comment by JerichoJones** [comment]
@guile124 Maybe I'm misunderstanding... you can't use the same AMD config across those algos.

**Comment by guile124** [comment]
@JerichoJones 
Exactly that, in xmr v7 I got it to work at about 400 h/s on average, in liteV7 (turtlecoin) it shows good hashrate report but in the results only get AMD invalid result gpu

**Comment by JerichoJones** [comment]
Is this resolved?

**Comment by Spudz76** [comment]
Not really, best solution for Terascale is to revert to fglrx 15.x, which means Ubuntu 14.
All the newer drivers do them wrong, either memory sizes or worse.
Apparently windows is OK also with 15.x drivers but I'd rather run Ubuntu 14.
Stats also don't always work right on old cards (Terascale, or say CUDA 20 platform) and I've been unable to trace why.  I have one that only shows speed in the second and third time columns but never ever ever in the 60s column (weird).  The HD5570 used to show a speed if you refreshed / hit H on console a bunch of times in a row (race condition).

There appears to be no real fix since the drivers are what is broken.

**Comment by Simaex** [comment]
@Spudz76 Do you get HD6990 actually mining under Ubuntu 14? If yes what performance do you get?
I was going to try 15.04   Vivid Vervet  with core  3.19 and 15.7 driver, the latest to support HD6990 but have not done it yet.
As of Windows it is no good to use 15.xx drivers. Much better results we have using 14.4 driver, something around 800 H/s per card (400 per GPU core).

**Comment by Simaex** [comment]
@Spudz76 I have read somewhere that the last kernel to properly support Terascale is 3.19
It seems that it is worth trying to check 4.xx kernels if I correctly understand that you have succesfully used 4.2 kernel for Terascale. How much memory is detected on the card? What worksize are you using at the moment? Strange but in Windows we have best results with a strange worksize of 19

**Comment by Spudz76** [comment]
I stopped using the Redwood mainly because I ran out of slots and the GTX550Ti and friends are faster.
Have to swap motherboards around to bring it back online to play with again.

Detects at 896MB and these are the last settings I was using:
```
  // gpu: Redwood memory:896
  // compute units: 5
  { "index" : 0,
//    "intensity" : 440, "worksize" : 8,
    "intensity" : 168, "worksize" : 3,
    "affine_to_cpu" : false, "strided_index" : 1, "mem_chunk" : 2,
    "comp_mode" : true
  },
```
The 440/8 is the autotuned default which gave no hashrate forever but kept running.  The other 168/3 was about the max I could do before losing hashrate back to null.  I think again this resulted in around 80 (76?) H/s.  OpenCL never crashes and never errors, just larger intensity never appears to work.

`fglrx-core, 15.201.2, 4.2.0-42-generic, x86_64: installed`
